src/main/scala/beam/agentsim/agents/modalbehaviors/ChoosesMode.scala

```diff
@@ -601,8 +514,7 @@ trait ChoosesMode {
     driveTransitTrip: Option[EmbodiedBeamTrip],
     rideHail2TransitResult: Option[RideHailResponse]
   ): Boolean = {
-    driveTransitTrip.isDefined && driveTransitTrip.get.legs
-      .exists(leg => beamScenario.rideHailTransitModes.contains(leg.beamLeg.mode)) &&
+    driveTransitTrip.exists(_.legs.exists(leg => beamScenario.rideHailTransitModes.contains(leg.beamLeg.mode))) &&
```

Comment:
I do not believe this to be better performance nor does it add much to readability. I guess it doesn't matter in the long run here, but I'd personally prefer to not change this one.